lib/mix/lib/mix/tasks/release.init.ex

```diff
@@ -104,7 +104,7 @@ defmodule Mix.Tasks.Release.Init do
     export RELEASE_BOOT_SCRIPT_CLEAN="${RELEASE_BOOT_SCRIPT_CLEAN:-"start_clean"}"
 
     rand () {
-      od -t xS -N 2 -A n /dev/urandom | tr -d " \n"
+      dd count=1 bs=2 if=/dev/urandom 2> /dev/null | od -x  | head -n1 | awk '{print $2}'
```

Comment:
Beware that this will change the output a little, as earlier version didn't printed newline at the end of output. Additionally I would use `cut` instead of `awk`:

```sh
dd count=1 bs=2 if=/dev/urandom 2> /dev/null | od -x  | head -n1 | cut -d' ' -f2
```

 or reduce amount of the commands ran via:

```sh
dd count=1 bs=2 if=/dev/urandom 2> /dev/null | od -x  | awk 'NR==1{printf("%s",$2)}'
```

Reply:
> Beware that this will change the output a little, as earlier version didn't printed newline at the end of output. 

Yes i saw that, but it seems to work on my pc and on the target (and is also used in relx).

> dd count=1 bs=2 if=/dev/urandom 2> /dev/null | od -x  | head -n1 | cut -d' ' -f2

This works on my linux pc, but not on the target using busybox. cut seems also be different there. So on the target this works (but it does not work on my pc)
```
dd count=1 bs=2 if=/dev/urandom 2> /dev/null | od -x  | head -n1 | cut -d' ' -f6
```

> dd count=1 bs=2 if=/dev/urandom 2> /dev/null | od -x  | awk 'NR==1{printf("%s",$2)}'

This works on both, the target and the pc.

Reply:
> Beware that this will change the output a little, as earlier version didn't printed newline at the end of output.

According to posix https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_06_03 newlines are removed at the end of command substitutions. This also works in busybox, so the additional `\n` at the end should not make a difference in this script.

Reply:
To sum this up, we have three choices here:
1. Close this pr and do not merge it -> mix releases do not work with busybox in a minimal config :worried:
2. Use the pr as it is now (see discussion above) :smile:
3. Change the pr to use `dd count=1 bs=2 if=/dev/urandom 2> /dev/null | od -x | awk 'NR==1{printf("%s",$2)}'` (see discussion above) :smile:

Reply:
Hi,

I'm in favor of option 3 or 2. When I sent the relx patch related to rand(), I did not explore the awk enough to get rid the head command.

Reply:
If newline at the end of output doesn't matter, then we can simplify the AWK call:

```
dd count=1 bs=2 if=/dev/urandom 2> /dev/null | od -x | awk 'NR==1{print $2}'
```

Reply:
@chrta does the command from @hauleth above work for you? If so, please update the PR and we can ship it!

Reply:
Very good, it works fine.